python/lsst/ci/prepare.py

```diff
@@ -486,10 +646,15 @@ def version(self, product_name, productdir, ref, dependencies):
         """
         q = pipes.quote
         cmd = "cd %s && pkgautoversion %s" % (q(productdir), q(ref))
-        product_version = subprocess.check_output(cmd, shell=True).decode("utf-8").strip()
-
+        process = await asyncio.create_subprocess_shell(cmd, stdout=subprocess.PIPE)
+        (stdout, stderr) = await process.communicate()
+        product_version = stdout.decode("utf-8").strip()
```

Comment:
Is it safer to use the default encoding here?

Reply:
🤔 git.py also uses utf-8 historically, but maybe that should be fixed too.

Reply:
From my reading, non-plumbing git commands (`git log` - see `--encoding`) seem to spit out UTF-8 by default. 

I'm not sure if that applies to pkgautoversion and the git commands used inside of it, but I think I'd prefer to leave it since the utf-8 decode was used in the previous implementation of this code.

Reply:
If all the decoding is for data from git then please consider a global constant, e.g. GIT_ENCODING or something like that. Though I would also be comfortable with removing the encoding and using the default.

I suspect the original code had to specify `utf-8` because Python 2's default encoding is ASCII, whereas Python 3 defaults to the desired `utf-8`.